### PR TITLE
Implement initial KfDef status

### PR DESCRIPTION
**Pull request by OchiengEd**
Implement initial KfDef status which will report if the reconciliation was successful based on if the KfApply function receives an error or not.

### CODE REVIEW

**Comment by nakfour** [comment]
@OchiengEd can you please post some testing steps. Thanks

File: pkg/controller/kfdef/status.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+		conditions = append(conditions, kfdefv1.KfDefCondition{
+			LastUpdateTime: cr.CreationTimestamp,
+			Status:         corev1.ConditionTrue,
+			Reason:         err.Error(),
+			Type:           kfdefv1.KfDegraded,
+		})
+	}
+
+	conditions = append(conditions, kfdefv1.KfDefCondition{
+		LastUpdateTime: cr.CreationTimestamp,
```
Comment (VaishnaviHire) [inline pkg/controller/kfdef/status.go:53]: Should we only update status if we get a new condition ? We do not update if the condition that we are about to add already exists and has the same status and reason.

**Review by VaishnaviHire — COMMENTED** [review]
Thanks @OchiengEd for working on this.  I just have a quick question

File: pkg/controller/kfdef/status.go (hunk 1 above)
Comment (OchiengEd) [reply pkg/controller/kfdef/status.go:53]: The actual comparison  between the current and previous conditions happens on [line 82](https://github.com/opendatahub-io/opendatahub-operator/blob/335ccd1a9796eda08c1983396b5eb0f6c1b7b264/pkg/controller/kfdef/status.go#L28). If the current status is empty or the new conditions are different from the existing ones, then the status would be updated.

**Comment by LaVLaS** [comment]
/lgtm

I tested this on top of the most recent changes.  The `kfdef` reported the successful status block below

```
conditions:
  - lastTransitionTime: null
    lastUpdateTime: '2021-04-21T15:39:47Z'
    reason: Kubeflow Deployment completed
    status: 'True'
    type: Available
```

A failed `kfdef` deployment reported 
```
conditions:
  - lastTransitionTime: null
    lastUpdateTime: '2021-04-21T19:04:51Z'
    reason: 'couldn''t generate KfApp:  (kubeflow.error): Code 500 with message: kfApp
      Generate failed for kustomize: Kustomize generate failed:  (kubeflow.error):
      Code 500 with message: application kubeflow-apps refers to repo foo-manifests
      which wasn''t found in KfDef.Status.ReposCache'
    status: 'True'
    type: Degraded
  - lastTransitionTime: null
    lastUpdateTime: '2021-04-21T19:04:51Z'
    reason: Kubeflow Deployment completed
    status: 'True'
    type: Available
```

**Review by LaVLaS — APPROVED** [review]
/lgtm

No issues. `.status` block was updated after initial rollout and any change from failed -> successful build

**Pull request merged**